intezer_sdk/analyses_results.py

```diff
@@ -0,0 +1,89 @@
+from typing import List, Any
+from typing import Dict
+from typing import Tuple
+
+from intezer_sdk.api import IntezerApi
+from intezer_sdk.api import raise_for_status
+
+
+class AnalysesResults:
+    def __init__(self, request_url_path: str, api: IntezerApi, request_filters: Dict):
+        """
+        All analyses results from server with the ability to iterate results.
+        :param request_url_path: Url to request new filter from.
+        :param api: Instance of Intezer API for request server.
+        :param request_filters: Filters requested from server.
+        """
+        self.api = api
+        self.request_filters = request_filters
+        self._pages = []
+        self._current_page: List[Any] = None
+        self._request_url_path = request_url_path
+        self._current_page_number = 0
+        self._total_count = 0
+        self._current_offset = 0
+
+    def __iter__(self):
+        """Iterate between page."""
+        if not self._current_page:
+            self._fetch_page()
+        if self._current_page:
+            yield from self._current_page
+            if len(self._pages) * self.request_filters['limit'] < self.total_count:
+                self._fetch_page()
+                yield from iter(self)
+
+    def __len__(self):
+        return len(self._pages)
+
+    @property
+    def total_count(self):
+        return self._total_count
+
+    def current_page(self) -> List:
+        """Get current page, if not exits, ask a new one from server."""
+        return self._current_page or self._fetch_page()
+
+    def all(self) -> List:
+        """Return all exits analysis's from server."""
+        self._fetch_all_pages()
```

Comment:
why aren't we using the results? if we take the results from the current state, just remove the `list(self)` so we won't create a list we don't use